torch/_inductor/fx_passes/micro_pipeline_tp.py

```diff
@@ -881,7 +904,7 @@ def fuse_matmul_reduce_scatter(reduce_scatter: _ReduceScatterMatch) -> None:
         return
 
     filter_matmul = None
-    if orig_scatter_dim == _get_tensor(input_node).ndim - 1:
+    if _is_last_dim(_get_tensor(input_node), orig_scatter_dim):
```

Comment:
Do you see the same small last dim len issue (< 1024) for fuse matmul reduce scatter?

Reply:
Yes, for autoparallel torchtitan I observed that with TP dim = 8 and sequence length = 4096, the decomposition was slower. With sequence length = 8192 (chunked size = 1024) the AsyncTp decomposed was +4-5%.